client/backstage-common/src/client/Query.ts

```diff
@@ -0,0 +1,100 @@
+import { gql } from 'graphql-request'
+import type { AuthenticatedUser } from '../../../shared/src/auth'
+import { currentAuthStateQuery } from '../../../shared/src/auth'
+
+export interface Query<T> {
+    gql(): string
+    vars(): string
+    Marshal(data: any): T[]
```

Comment:
```suggestion
    marshal(data: any): T[]
```

We don't start method names with capital letters in TS 😅

Reply:
Fixed!